### ISSUE TITLE
Building a repeated schematic, it often skips the last block

### ISSUE THREAD

**Issue by graemeg** [cabaletta/baritone#2779]
## Some information
Operating system: Linux 18.04.5
Java version: 8 (282)
Minecraft version: 1.12.3
Baritone version: 1.2.15
Other mods (if used):  Kami Blue, Forge Mod Loader

## Exception, error or logs
There is no error in the log, simple odd behaviour in the game.

## How to reproduce
Any idea why Baritone building a schematic often skips the last block. In my case it's the top right block in the image - errors indicated with the Magenta circles. My schematic is to built diagonal highways and the schematic is a 1x4x7 (width x height x depth). Basically 7 blocks of Netherrack on the bottom layer, and One block of Netherrack on each end on layer two. The rest is air. I use buildrepeat, but  as you can see from the screenshot, often the top right block is left behind, when it should have been Air. I'm using Baritone 1.2.15 release version from Github, running on Minecraft 1.12.2 on the Constantiam server.

In the nether, digging a highway in NE direction.

```
buildrepeat 1,0,-1
build diagonal-dig-medium-23wide.schematic ~ 118 ~
```

Even if I disable the Kami Blue mod, it still does the same. So Kami Blue is not affecting the outcome.

![2021-05-28_19 40 49](https://user-images.githubusercontent.com/72053/120030995-1167be80-bff0-11eb-9ecb-60f33005d178.png)

Here is the schematic file.

[diagonal-dig-medium-23wide.schematic.zip](https://github.com/cabaletta/baritone/files/6562880/diagonal-dig-medium-23wide.schematic.zip)

## Modified settings

- rightClickSpeed: 2
- buildRepeat: 1,0,-1
- allowSprint: false
- blockReachDistance: 4.5
- useSwordToMine: false

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by graemeg** [comment]
I did some more experimentation. It works perfectly in a Single Player world, but not on Constantiam which experiences lower TPS values due to the number of players. I also tried using an unenchanted Iron pickaxe vs a Diamond Efficiency 5 pickaxe. On Constantiam, the results are the same, regardless what pickaxe I used. So I assume it's the lag that affects Baritone to think it mined the top right block, even though it did not break in time. :man_shrugging:

**Comment by ZacSharp** [comment]
Yes, lag is likely the reason.
In that case you can work around it by simply making your schematic longer.

**Comment by graemeg** [comment]
Is `rightClickSpeed` the only why to control the speed? Would it help if I make this setting larger when the TPS is not great? ie: Can in slow down the mining for me?

**Comment by ZacSharp** [comment]
No, there is no way to slow mining down, because Baritone just presses your left mouse button until the block disappears. 

The problem is that the server places the block back after a few moments and Baritone has already moved the schematic at that point. The only solution to this problem would be checking again after some time, just like a human would do.
